os/tools/mkconfig.c

```diff
@@ -205,21 +205,15 @@ int main(int argc, char **argv, char **envp)
 	printf("# define CONFIG_SMARTFS_JOURNALING_THRESHOLD (CONFIG_SMARTFS_NLOGGING_SECTORS * 4)\n");
 	printf("#endif\n\n");
 	printf("/* There must be at least one memory region. */\n\n");
-	printf("#ifndef CONFIG_MM_REGIONS\n");
-	printf("# define CONFIG_MM_REGIONS 1\n");
-	printf("#endif\n\n");
-	printf("#ifdef CONFIG_MM_KERNEL_HEAP\n");
 	printf("#ifndef CONFIG_KMM_REGIONS\n");
 	printf("# define CONFIG_KMM_REGIONS 1\n");
-	printf("#endif\n");
-	printf("#if CONFIG_KMM_REGIONS > CONFIG_MM_REGIONS\n");
-	printf("#define CONFIG_MM_REGION_NUM CONFIG_KMM_REGIONS\n");
-	printf("#else\n");
-	printf("#define CONFIG_MM_REGION_NUM CONFIG_MM_REGIONS\n");
-	printf("#endif\n");
+	printf("#endif\n\n");
+	printf("#ifdef __KERNEL__\n");
+	printf("# define CONFIG_MM_REGIONS CONFIG_KMM_REGIONS\n");
 	printf("#else\n");
-	printf("#define CONFIG_MM_REGION_NUM CONFIG_MM_REGIONS\n");
-	printf("#endif\n");
+	printf("# define CONFIG_MM_REGIONS 1\n");
```

Comment:
You removed the `CONFIG_MM_REGIONS` and `CONFIG_MM_NHEAPS`. Are these needed?

Reply:
Fixed